framework/Web/UI/TTheme.php

```diff
@@ -331,4 +337,29 @@ protected function setJavaScriptFiles($value)
 	{
 		$this->_jsFiles = $value;
 	}
+	
+	/**
+	 * This post-processes the theme for RTL.  It loops through all the
+	 * css files, removes all the '.*rtl.css' and '.*rtl.(media).css' files,
+	 * then if there is globalization and globalization is RTL then re-adds
+	 * the rtl css at the end.
+	 * @since 4.2.0
+	 */
+	protected function postProcessCssRTL()
+	{
+		$rtlCss = [];
+		foreach ($this->_cssFiles as $key => $url) {
+			$segs = explode('.', basename($url));
+			if ((substr_compare(strtolower($url), 'rtl.css', -7) === 0) ||
+				(isset($segs[2]) && (substr_compare(strtolower($segs[count($segs) - 3]), 'rtl', -3) === 0))) {
+				$rtlCss[] = $url;
+				unset($this->_cssFiles[$key]);
+			}
+		}
+		if ($globalization = $this->getApplication()->getGlobalization()) {
+			if ($globalization->getIsCultureRTL()) {
+				$this->_cssFiles = array_merge($this->_cssFiles, $rtlCss);
+			}
+		}
```

Comment:
This is a bit overly complicated, removing things from an array to add them back later.
I would rework it (in pseudocode):
```
// rtl: no css needs to be removed
if($globalization = ... && $globalization->getIsCultureRTL())
  return;

// remove rtl css files
foreach($this->_cssFiles as $key => $url) {
	if($this->getIsRtlCssFile($url)) {
	  unset($this->_cssFiles[$key]);
	}
}

```

Also, moving the actual comparison to an external "getIsRtlCssFile()" method would clarify its meaning

Reply:
My thinking on this that LTR should first have RTL css removed by default.  This way LTR are protected from RTL css enabled themes.  Automatically including RTL (when there is no globalization) would always make themes RTL by default if the theme supports RTL, that's not preferred behavior.
Then if there is globalization and it's RTL, then the CSS is added _at the end_ of all css.  The "rtl css at the end" aspect is important for CSS priority of RTL.  The files later would override the LTR css with redefinitions. Without being re-added at the end would mean the RTL css would be based on file name order rather than having RTL css after all LTR css.

I am ambivalent on a getIsRtlCssFile method.  If you think it'll be good, I'll get on it.

Reply:
You have a good point on this, i missed that.
You can keep it as it is.